proposals/0200-raw-string-escaping.md

```diff
@@ -1,46 +1,432 @@
+   # static-string-literal #
+```
 
-## Detailed design
+Any instance of the delimiter which is not followed by the appropriate number of pound signs is treated as literal string contents, rather than as the end of the string literal. That is, the leading pound signs *change the string's delimiter* from `"` to `"#` (or `"##`', etc.). A plain `"` without pound signs after it is just a double-quote character inside the string.
 
-The changes are confined to the file lib/Parse/Lexer.cpp and involves check for a flag whether the string was prefixed by r disabling processing processing of backslash escapes and is localised to the function Lexer::lexCharacter in the code. If the "r" introducer was present, both the backslash and the unknown character are passed into the literal in Lexer::getEncodedStringSegment. A further minor change is also required to the main switch statement in Lexer::lexImpl and Token.h to convey the flag from parsing to code generation phases.
+```
+#"She said, "This is dialog!""#
+// Equivalent to "She said, \"This is dialog!\""
+```
+
+If you do add a backslash, it is interpreted as an extra character. This string literal includes the backslash and both double quote marks inside the string delimiters (`#"` and `"#`):
+
+```
+#"A \"quote"."#
+```
+
+When you need to include `#"` (pound-quote) or `"#` (quote-pound) in your character sequence, adjust the number of delimiting pound signs. This need should be rare.
+
+### Customized Escape Delimiters
+
+This design uses an *escape delimiter*, that is a sequence of one or more characters to indicate the beginning of an escape character sequence, rather than a single escape character.  Like Swift today, the escape delimiter begins with a backslash (Reverse Solidus, U+005C), but it is now followed by zero or more pound signs
+(Number Sign, U+0023). An escape delimiter in a string literal must match the number of pound signs used to delimit either end of the string.
+
+Here is the degenerate case. It is a normal string with no pound signs. The escape delimiter therefore needs no pound signs and a single backslash is sufficient to establish the escape character sequence:
+
+```
+"This string has an \(interpolated) item"
+```
+
+Strings using custom boundary delimiters mirror their pound sign(s) after the leading backslash, as in these examples which produce identical results to the preceding string literal:
+
+```
+#"This string has an \#(interpolated) item"#
+
+####"This string has an \####(interpolated) item"####
+```
+
+The escape delimiter customization matches the string. Any backslash that is not followed by the correct number of pound signs is treated as raw text. It is not an escape:
+
+```
+#"This is not \(interpolated)"#
+```
+
+| String Start Delimiter | Escape Delimiter | String End Delimiter |
+| ---------------------- | ---------------- | -------------------- |
+| `"` | `\` | `"` |
+| `#"` | `#\` | `"#` |
+| `##"` | `##\` | `"##` |
+| `######"` | `######\` | `"######` |
```

Comment:
In this table, shouldn't the escape delimiters be reversed?

- `\#` instead of `#\`
- `\##` instead of `##\`
- `\######` instead of `######\`

Putting the pound signs *before* the backslash also makes sense, but doesn't match the rest of the proposal.